javax-mail-extension/src/main/java/org/apache/james/javax/MimeMessageSaver.java

```diff
@@ -0,0 +1,36 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.javax;
+
+import java.util.Optional;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+import com.github.fge.lambdas.Throwing;
+
+public class MimeMessageSaver {
+    public static void save(MimeMessage mimeMessage) throws MessagingException {
+        String messageID = mimeMessage.getMessageID();
+        mimeMessage.saveChanges();
+        Optional.ofNullable(messageID)
+            .ifPresent(Throwing.consumer((String id) -> mimeMessage.setHeader("Message-ID", id)).sneakyThrow());
```

Comment:
in this case, the message has been half-changed but the client can continue to use it, right ?